pandas/sparse/array.py

```diff
@@ -191,20 +189,18 @@ def __new__(cls, data, sparse_index=None, index=None, kind='integer',
         if dtype is not None:
             dtype = np.dtype(dtype)
         is_sparse_array = isinstance(data, SparseArray)
-        if fill_value is None:
-            if is_sparse_array:
-                fill_value = data.fill_value
-            else:
-                fill_value = nan
 
         if is_sparse_array:
             sparse_index = data.sp_index
-            values = np.asarray(data)
+            values = data.sp_values
+            fill_value = data.fill_value
         else:
             # array-like
             if sparse_index is None:
-                values, sparse_index = make_sparse(data, kind=kind,
-                                                   fill_value=fill_value)
+                if dtype is not None:
```

Comment:
prob need to have this handled for ``.astype`` as well. you already have something like this getting the fill value, maybe can use that to validate. (e.g. use the if test, then raise if its not there).